app/uk/gov/hmrc/tariffclassificationfrontend/service/CasesService.scala

```diff
@@ -19,34 +19,34 @@ package uk.gov.hmrc.tariffclassificationfrontend.service
 import javax.inject.{Inject, Singleton}
 import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.tariffclassificationfrontend.connector.BindingTariffClassificationConnector
-import uk.gov.hmrc.tariffclassificationfrontend.models.{Case, CaseStatus, Queue}
+import uk.gov.hmrc.tariffclassificationfrontend.models._
+import uk.gov.hmrc.tariffclassificationfrontend.models.request.NewEventRequest
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
 @Singleton
 class CasesService @Inject()(connector: BindingTariffClassificationConnector) {
 
-  def releaseCase(c: Case, queue: Queue)(implicit hc: HeaderCarrier): Future[Case] = {
-    // TODO: DIT-246 - with an atomic operation we should:
-    // - update status
-    // - create case status change event
-    // If too complex, we should at least update the event and the event atomically
-    connector.updateCaseStatus(caseReference = c.reference, newStatus = CaseStatus.OPEN).flatMap { withNewStatus: Case =>
-      connector.updateCase(withNewStatus.copy(queueId = Some(queue.id)))
-    }
+  def releaseCase(c: Case, queue: Queue)(implicit hc: HeaderCarrier, operator: Operator): Future[Case] = {
+    val eventualCase: Future[Case] = connector.updateCase(c.copy(status = CaseStatus.OPEN, queueId = Some(queue.id)))
+    eventualCase.onSuccess({
```

Comment:
Just use `map` and use one line of code

Reply:
Map feels wrong, you're not actually mapping it to anything

Reply:
It's a shame `onSuccess` returns `Unit`